clients/privacy-center/components/consent/ConfigDrivenConsent.tsx

```diff
@@ -59,6 +64,13 @@ const ConfigDrivenConsent = ({
         consentOption: option,
         consentContext,
       });
+      option.cookieKeys?.forEach((cookieKey) => {
+        const previousConsent = cookieKeyConsent[cookieKey];
+        // For a cookie key to have consent, _all_ data uses that target that cookie key
+        // must have consent.
+        cookieKeyConsent[cookieKey] =
+          previousConsent === undefined ? value : previousConsent && value;
```

Comment:
hm how does this work if the `previousConsent` was `true` but now they are setting it to `false`? wouldn't the `previousConsent && value` make this still persist as `true`?

Reply:
are we not able to call `makeCookieKeyConsent` here? I imagine we are trying to move where we create the cookie from the `useEffect` in `consent.tsx` to here, so are we able to reuse that function?

Reply:
oh doh you're right, I don't know my boolean logic!!!